frontend/src/metabase/query_builder/components/notebook/lib/steps.ts

```diff
@@ -44,21 +44,24 @@ const STEPS: NotebookStepDef[] = [
       const database = query.database();
       return query.hasData() && database != null && database.hasFeature("join");
     },
-    subSteps: query => query.joins().length,
-    active: (query, index) =>
-      typeof index === "number" && query.joins().length > index,
-    revert: (query, index) => query.removeJoin(index),
-    clean: (query, index) => {
-      const join = typeof index === "number" ? query.joins()[index] : null;
-      if (!join || join.isValid() || join.hasGaps()) {
-        return query;
+    subSteps: (topLevelQuery, stageIndex) =>
```

Comment:
The second code snippet adds unnecessary complexity by introducing an additional function Lib.joins() and requiring more parameters for the functions.